### DISCUSSION TITLE
Batch process video files

### DISCUSSION THREAD

**Discussion by QuentinGG** [chaiNNer-org/chaiNNer#2955] [Ideas]
Hi,
First of all, thanks a lot for creating this amazing tool! Super useful in a production pipeline environnent!

I noticed that we can specify a directory as input for images but not videos ?
This would be amazing for batch processing a directory's content.
Apologies if this is already implemented in another way, I couldn't find it...

Example of use:
I have a large list of ProRes mov source files and I need to batch proccess them into h264 outputs while saving the results folders named after the source files with some text replacement.  I can already do all of this,... but I have to do it file by file...

Thanks!

**Comment by joeyballentine** [comment]
This is currently a technical limitations of chainner. Since video files batch process their frames already, we can't batch process videos since that would be a batch of a batch. Once we figure out how to solve nested batching then this will be possible, but for now it's not.

**Reply by Wetzer2** [reply]
> This is currently a technical limitations of chainner. Since video files batch process their frames already, we can't batch process videos since that would be a batch of a batch. Once we figure out how to solve nested batching then this will be possible, but for now it's not.

What if:
* Node LOAD VIDEO would be reduced to only load video file with video stream and audio stream output (and of course other things like NAME, DIRECTORY, etc).
* create new node SPLIT VIDEO STREAM with the function of split video stream in image sequence.
* add to node SAVE VIDEO the OR function for input video stream or image sequence

After that, you can use LOAD VIDEO as the same of LOAD IMAGES and you can link it also directly to SAVE VIDEO. If you want work with specific images from the video therefore you have SPLIT VIDEO STREAM.

**Reply by joeyballentine** [reply]
That wouldn't solve what OP wants though

**Reply by QuentinGG** [reply]
Thanks for the clairification about why it's not currently possible. 
I'll be waiting for this feature :)
